### ISSUE TITLE
Cannot recover account NAME

### ISSUE THREAD

**Issue by JobLeonard** [mozilla/thimble.mozilla.org#1552]
So I have a bit of a weird problem:  I apparently signed up a long time ago and I forgot my username. So I click "reset password", fill in my email, get a link sent to my email account to reset my password. However, there is no mention of my username, and I cannot log in with my email address either. So I'm stuck.

Please send the account name to the email address, and/or let people log in with their email address.

**Comment by gideonthomas** [comment]
cc @cadecairos

**Comment by cadecairos** [comment]
A while back, someone decided that logging in via username AND email was "too hard on users" .... and look where that got us. Funny thing is, the API allows for log in via username too, someone just decided to forbid it client-side!

@JobLeonard I'd be glad to help you recover your username in the mean time. Could you send an email to me (you can find my Mozilla Foundation email on my [GitHub Profile](https://github.com/cadecairos))? I can assist you from there.

**Comment by JobLeonard** [comment]
>  username AND email was "too hard on users" 

Well, I agree with that, but that's why most websites have the option to use a username OR email ;). Although I suppose that might be a security risk of sorts, statistically speaking.

Perhaps adding the username as a reminder  in the recovery email would be a nice compromise? You could then just change the link to "forgot username or password"

I will send you an email, thank you for your swift response!

**Comment by cadecairos** [comment]
> Perhaps adding the username as a reminder in the recovery email would be a nice compromise?

Yeah, I think this is the easiest way to ensure we don't need people to remember the username they chose.
